staging/src/k8s.io/client-go/tools/cache/listers_test.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	schema "k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestListersListAll(t *testing.T) {
+	mkObj := func(id string, val string) testStoreObject {
+		return testStoreObject{id: id, val: val}
+	}
+
+	store := NewStore(testStoreKeyFunc)
+
+	err := store.Add(mkObj("foo", "bar"))
+	if err != nil {
+		t.Errorf("store obj add failed")
+	}
+
+	err = store.Add(mkObj("foo-1", "bar-1"))
+	if err != nil {
+		t.Errorf("store obj add failed")
+	}
+
+	expectedOutput := []testStoreObject{
+		mkObj("foo", "bar"),
+		mkObj("foo-1", "bar-1"),
+	}
+	actualOutput := []testStoreObject{}
+
+	err = ListAll(store, labels.Everything(), func(obj interface{}) {
+		actualOutput = append(actualOutput, obj.(testStoreObject))
+	})
+
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if !reflect.DeepEqual(expectedOutput, actualOutput) {
```

Comment:
is the ListAll guaranteed to return the objects in the same order?

Reply:
Yes, The sort order is lexographical by name.  Kube simply retains the order etcd provides from it's keyspace, so it's a defacto ordering that is backed into the API to the point where we would never change it.  We do depend on this order for things like [--limit](https://kubernetes.io/docs/reference/using-api/api-concepts/#retrieving-large-results-sets-in-chunks).